pkg/services/ngalert/tests/fakes/rules.go

```diff
@@ -0,0 +1,341 @@
+package fakes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	models2 "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+// FakeRuleStore mocks the RuleStore of the scheduler.
+type FakeRuleStore struct {
+	t   *testing.T
+	mtx sync.Mutex
+	// OrgID -> RuleGroup -> Namespace -> Rules
+	Rules       map[int64][]*models.AlertRule
+	Hook        func(cmd interface{}) error // use Hook if you need to intercept some query and return an error
```

Comment:
Let's file a ticket to delete this, too. The tests that use it would be better served by a Fake whose implementation of `GetAlertRulesBy...` could be set to return an error than a Hook that uses type assertions.

Reply:
This one is simple enough, I added it as a task on the current refactor.